apps/src/applab/designMode.js

```diff
@@ -840,7 +840,7 @@ function makeDraggable(jqueryElements) {
         });
 
         // Dim the element if it's dragged out of bounds
-        if (!isMouseInBounds(newLeft, newTop)) {
+        if (!isMouseInBounds(newLeft + event.offsetX, newTop + event.offsetY)) {
```

Comment:
Good call, I've renamed isMouseInBounds to isPointInBounds. 

Regarding newLeft and newTop, in looking more closely I've realized that this approach does not work very well. This is primarily because jQuery.MouseEvent.offsetLeft doesn't account for the border of the element, so the region in which an image element (which has a 5px border) is in bounds is shifted 5px down and right when compared with the region in which a button element (which has no border) is in bounds.

I've taken a new approach of using event.clientX/Y plus scaling to compute whether the mouse is in bounds, and added a new isMouseEventInBounds helper.

Now that newLeft and newTop are not part of the equation, it stops mattering whether we do this check before or after we snap those values to the grid.